rmw_connext_cpp/src/rmw_init.cpp

```diff
@@ -12,15 +12,76 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include "rmw/rmw.h"
+#include "rmw/init.h"
 
+#include "rmw/impl/cpp/macros.hpp"
 #include "rmw_connext_shared_cpp/init.hpp"
 
+#include "rmw_connext_cpp/identifier.hpp"
+
 extern "C"
 {
 rmw_ret_t
-rmw_init()
+rmw_init_options_init(rmw_init_options_t * init_options, rcutils_allocator_t allocator)
+{
+  RMW_CHECK_ARGUMENT_FOR_NULL(init_options, RMW_RET_INVALID_ARGUMENT);
```

Comment:
Again, could we create shared functions to be used by connext and connext_dynamic for all of these?